dali/operators/segmentation/random_object_bbox.cc

```diff
@@ -341,16 +392,17 @@ bool RandomObjectBBox::PickBlob(SampleContext &ctx, int nblobs) {
   return false;
 }
 
-template <typename T>
+template <typename BlobLabel, typename T>
 bool RandomObjectBBox::PickForegroundBox(
-      SampleContext &context, const InTensorCPU<T> &input) {
+      SampleContext<BlobLabel> &context, const InTensorCPU<T> &input) {
   GetBgFgAndWeights(context.classes, context.weights, context.background, context.sample_idx);
   context.class_label = context.background;
+  auto &tp = *context.thread_pool;
   if (ignore_class_) {
-    int nblobs = LabelConnectedRegions(context.blobs, input, -1, context.background);
+    int nblobs = LabelConnectedRegions(context.blobs, input, tp, -1, context.background);
     return PickBlob(context, nblobs);
   } else {
-    FindLabels(context.labels, input);
+    FindLabels(context, input);
```

Comment:
Just a nitpick, FindLabels gets a context, FilterByLabel gets buffers and a thread pool.